pkg/telegram-bot/telegram-bot.go

```diff
@@ -1 +1,421 @@
+		selector.Row(refresh),
+		selector.Row(back),
+	)
+
+	return c.Edit(
+		log,
+		selector,
+	)
+}
+
+func myBookmarkBtn(c telebot.Context) error {
+	var (
+		prevPageMyBkMark = selector.Data("Previuse page", "Previuse_page_my_bookmark")
+		nexPageMyBkMark  = selector.Data("Next page", "Next_page_my_bookmark")
+	)
+	page := selector.Data(fmt.Sprintf("page %v of %v", 1, 1), "page")
+	bookMarks := fmt.Sprintf("My bookmark  : \n%s", "bookmark ..")
+	selector.Inline(
+		selector.Row(prevPageMyBkMark, page, nexPageMyBkMark),
+		selector.Row(globalBookmark),
+		selector.Row(back),
+	)
+
+	return c.Edit(
+		bookMarks,
+		selector,
+	)
+}
+
+func globalBookmarkBtn(c telebot.Context) error {
+	var (
+		prevPageGlobBkMark = selector.Data("Previuse page", "Previuse_page_global_bookmark")
+		nexPageGlobBkMark  = selector.Data("Next page", "Next_page_global_bookmark")
+	)
+	page := selector.Data(fmt.Sprintf("page %v of %v", 1, 1), "page")
+	bookMarks := fmt.Sprintf("global bookmark : \n%s", "bookmark ..")
+	selector.Inline(
+		selector.Row(prevPageGlobBkMark, page, nexPageGlobBkMark),
+		selector.Row(myBookmark),
+		selector.Row(back),
+	)
+
+	return c.Edit(
+		bookMarks,
+		selector,
+	)
+}
+
+func postBtn(c telebot.Context) error {
+	var (
+		prevPagePost = selector.Data("Previuse page", "Previuse_page_post")
+		nexPagePost  = selector.Data("Next page", "Next_page_post")
+	)
+	page := selector.Data(fmt.Sprintf("page %v of %v", 1, 1), "page")
+	posts := "Select a repost to display its posts : "
+
+	// copule of inline btn .
+
+	selector.Inline(
+		selector.Row(prevPagePost, page, nexPagePost),
+		selector.Row(back),
+	)
+
+	return c.Edit(
+		posts,
+		selector,
+	)
+}
+
+func newPostBtn(c telebot.Context) error {
+	posts := "Enter a name for your post : "
+
+	// copule of inline btn .
+
+	selector.Inline(
+		selector.Row(newPost),
+		selector.Row(back),
+	)
+
+	return c.Edit(
+		posts,
+		selector,
+	)
+}
+
+func reportBtn(c telebot.Context) error {
+	var (
+		prevPageReport = selector.Data("Previuse page", "Previuse_page_report")
+		nexPageReport  = selector.Data("Next page", "Next_page_report")
+	)
+	page := selector.Data(fmt.Sprintf("page %v of %v", 1, 1), "page")
+	report := fmt.Sprintf("The report \"%s\" has been selected , select a post to view , edit or delere :", "vespa 200 divar and sheypoor")
+
+	// copule of inline btn .
+
+	selector.Inline(
+		selector.Row(prevPageReport, page, nexPageReport),
+		selector.Row(newPost),
+		selector.Row(back),
+	)
+
+	return c.Edit(
+		report,
+		selector,
+	)
+}
+
+func backBtn(c telebot.Context) error {
+	selector.Inline(
+		selector.Row(profile, bookmark, monitoring),
+		selector.Row(post, export, report),
+		selector.Row(logout),
+	)
+
+	return c.Edit(
+		fmt.Sprintf("Welcome dear %s %s", c.Chat().FirstName, c.Chat().LastName),
+		selector,
+	)
+}
+
+func profileBtn(c telebot.Context) error {
+	c.Delete()
+	return c.Send("Hello mr mohammadi !")
+}
+
+func onText(c telebot.Context) error {
```

Comment:
reading this function is really hard, please make it sample by creating smaller functions